### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by pearofducks**
- Fixes many 404s on the Fastify.io site (and preserves working links in GitHub repo browsing)
- Fixes or improves a couple other pieces of documentation

Fixes: https://github.com/fastify/fastify/issues/2421
Fixes: https://github.com/fastify/website/issues/177
Fixes: https://github.com/fastify/website/issues/178
Fixes: https://github.com/fastify/website/issues/179
Fixes: https://github.com/fastify/website/issues/181

### CODE REVIEW

**Comment by iandrc** [comment]
@pearofducks could you add the following to the PR description:
```
Fixes: https://github.com/fastify/fastify/issues/2421
Fixes: https://github.com/fastify/website/issues/177
Fixes: https://github.com/fastify/website/issues/178
Fixes: https://github.com/fastify/website/issues/179
Fixes: https://github.com/fastify/website/issues/181
```

In this case, GitHub will autoclose those related issues

File: docs/ContentTypeParser.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ Natively, Fastify only supports `'application/json'` and `'text/plain'` content
 
 As with the other APIs, `addContentTypeParser` is encapsulated in the scope in which it is declared. This means that if you declare it in the root scope it will be available everywhere, while if you declare it inside a plugin it will be available only in that scope and its children.
 
-Fastify automatically adds the parsed request payload to the [Fastify request](./Request.md) object which you can access with `request.body`.
```
Comment (iandrc) [inline docs/ContentTypeParser.md:8]: Does it brake the navigation between the different pages in IDEs, GitHub, and similar stuff? I'm a bit concerned that those changes may break this.

**Comment by iandrc** [comment]
@Eomm could you take a look?

File: docs/ContentTypeParser.md (hunk 1 above)
Comment (pearofducks) [reply docs/ContentTypeParser.md:8]: I assume you mean does changing from the `./Request.md` to `Request.md`.

There's already a page on `master` with this style.

The first link to Plugins on [this page](https://github.com/fastify/fastify/blob/master/docs/Decorators.md) uses this style of link, and doesn't 404 on GitHub or the Fastify site.

Comment (iandrc) [reply docs/ContentTypeParser.md:8]: Yeah, that was my question. 

I've checked it. Looks good to me, let's wait for others to review :)

**Comment by Eomm** [comment]
I don't have a pc for the next few days. I looked some pages and it seems to work on github without the ./

We could just build the site locally with these md files to check if it fixes the links there also

**Comment by pearofducks** [comment]
@Eomm - The link to Plugins (`To learn more, see the Plugins documentation.`) on [this page](https://www.fastify.io/docs/latest/Decorators/) already has this format of link. So we can validate on the live site already that changing the link format works. :)

**Comment by iandrc** [comment]
@pearofducks would you mind to squash your commits into 1?

**Comment by pearofducks** [comment]
@iandrc - can't you Squash and Merge when merging this PR?

I'm ofc happy to - I just left it as 3 changes so it was easier to visualize the 404 fixes vs. other doc improvements.

**Comment by iandrc** [comment]
@pearofducks thanks for squashing :)

**Pull request merged**